### DISCUSSION TITLE
Issue with multiple ScenegraphLayers only rendering last GLB model in layer stack

### DISCUSSION THREAD

**Discussion by cramatt** [visgl/deck.gl#6087] [Q&A]
I have a visualization where I'm trying to represent 10 different moving vehicles and each uses a different GLB model. I'm using 10 `ScenegraphLayer`s each with a different `scenegraph` property and a single `data` property representing the vehicles current position. Something like

```jsx
const layers = data.map((vehicle, i) => {
    return new ScenegraphLayer({
      id: vehicle.id,
      data: [vehicle],
      scenegraph: vehicle.glbUrl,
      getPosition: () => {
        ...
        return [xPos, yPos, zPos];
      }
    })
  });

<DeckGL
  ...
  layers={layers}
 />
```

Very similar to this example except each aircraft would have a model that represents the aircraft type, like plane, helicopter, etc. https://github.com/visgl/deck.gl/blob/8.5-release/examples/website/scenegraph/app.js 

First, is this the correct approach for rendering multiple GLB files?

Second, I seem to be encountering a bug where only the last SceneGraphLayer is rendering the model. It's always the last one in the layer stack. Is is possible this is a limitation of `ScenegraphLayer`?

**Comment by jonathanperezg** [comment]
> I have a visualization where I'm trying to represent 10 different moving vehicles and each uses a different GLB model. I'm using 10 `ScenegraphLayer`s each with a different `scenegraph` property and a single `data` property representing the vehicles current position. Something like
> 
> ```js
> const layers = data.map((vehicle, i) => {
>     return new ScenegraphLayer({
>       id: vehicle.id,
>       data: [vehicle],
>       scenegraph: vehicle.glbUrl,
>       getPosition: () => {
>         ...
>         return [xPos, yPos, zPos];
>       }
>     })
>   });
> 
> <DeckGL
>   ...
>   layers={layers}
>  />
> ```
> 
> Very similar to this example except each aircraft would have a model that represents the aircraft type, like plane, helicopter, etc. https://github.com/visgl/deck.gl/blob/8.5-release/examples/website/scenegraph/app.js
> 
> First, is this the correct approach for rendering multiple GLB files?
> 
> Second, I seem to be encountering a bug where only the last SceneGraphLayer is rendering the model. It's always the last one in the layer stack. Is is possible this is a limitation of `ScenegraphLayer`?

Hello, did you ever find a solution for this? I am encountering a similar problem. Are you trying to load the .glb file from a local directory?
